src/pydata_sphinx_theme/__init__.py

```diff
@@ -1044,7 +1048,7 @@ def setup(app):
     app.set_translator("readthedocs", BootstrapHTML5Translator, override=True)
     app.set_translator("readthedocsdirhtml", BootstrapHTML5Translator, override=True)
 
-    app.connect("env-updated", update_config)
+    app.connect("builder-inited", update_config)
```

Comment:
the fact that you now intercept "builder-inited" instead of "env-update" means that you take into account default values from `theme.conf`. 

Does it means that verything that is done in [`prepare_html_config`](https://github.com/pydata/pydata-sphinx-theme/blob/5fc277429ce0e8843d9bea4a84defcc60200bcef/src/pydata_sphinx_theme/__init__.py#L161) could be moved to [`update_config`](https://github.com/pydata/pydata-sphinx-theme/blob/5fc277429ce0e8843d9bea4a84defcc60200bcef/src/pydata_sphinx_theme/__init__.py#L32) ?

Reply:
I do think that this is the "cleaner" way to intercept and/or set our own defaults for config. Happy to explore moving that functionality to this event if you think that'd be useful

Reply:
I think it would make sense in a folow-up PR but great that you found out this event !

Reply:
A quick follow-up here. I actually don't think that we can move the theme-related deprecations into the `builder-inited` event.

This is because `builder-inited` (by its name) comes _after the HTML builder has been inited_, and at that point the theme's configuration structure has already been finalized. We can then alter that structure later on in the `html-page-context` event, but if we alter it in `app.config` earlier on, it will have no effect.